modules/latex.php

```diff
@@ -19,88 +19,464 @@
+			default:
+				return false;
+		}
+	}
+
+	/**
+	 * Print a valid MathJax expression.
+	 *
+	 * Wraps LaTeX expression with LaTeX font size command sequence when necessary.
+	 * Wraps expression in a span to control colors when necessary.
+	 *
+	 * @since 1.1
+	 *
+	 * @param  string  $latex LaTeX expression.
+	 * @param  string  $fg    Foreground color.
+	 * @param  string  $bg    Background color.
+	 * @param  integer $s     Font size. Default '0'.
+	 * @return string         LaTeX expression.
+	 */
+	public function render( $latex, $fg, $bg, $s = 0 ) {
+		// Output MathJax if it's not already on the page.
+		$this->output_mathjax_script();
+
+		$latex_size_pt   = $this->latex_size_pt( $s );
+		$styles          = array();
+		$text_mode_value = $this->mathjax_text_mode_workarounds( trim( $latex ), $fg, $bg, $s );
+
+		if ( $fg ) {
+			$styles[] = 'color: #' . $fg . ';';
+		}
+		if ( $bg ) {
+			$styles[] = 'background-color: #' . $bg . ';';
+		}
+		if ( $latex_size_pt ) {
+			$styles[] = 'font-size: ' . $latex_size_pt . ';';
+		}
 
-add_filter( 'no_texturize_shortcodes', 'latex_no_texturize' );
+		if ( $text_mode_value ) {
+			$output   = $text_mode_value;
+			$styles[] = 'line-height: 0;';
+		} else {
+			$output = sprintf('$latex %s$', $latex);
+		}
 
-add_filter( 'the_content', 'latex_markup', 9 ); // before wptexturize
-add_filter( 'comment_text', 'latex_markup', 9 ); // before wptexturize
-add_shortcode( 'latex', 'latex_shortcode' );
+		if ( $styles ) {
+			$output = sprintf('<span style="display: inline-block;%s">%s</span>',
+				implode('', $styles),
+				$output
+			);
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Render a LaTeX expression using WP Latex.
+	 *
+	 * To use for edge cases where MathJax can't render something.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @param  string  $latex LaTeX expression.
+	 * @param  string  $fg    Foreground color.
+	 * @param  string  $bg    Background color.
+	 * @param  integer $s     Font size.
+	 * @return string
+	 */
+	public function render_img( $latex, $fg, $bg, $s = 0 ) {
+		$url = "//s0.wp.com/latex.php?latex=" . urlencode( $latex ) . "&bg=" . $bg . "&fg=" . $fg . "&s=" . $s;
+		$url = esc_url( $url );
+		list($width, $height) = getimagesize('http:' . $url);
+		$alt = str_replace( '\\', '&#92;', esc_attr( $latex ) );
+		return '<img src="' . $url . '&zoom=2" alt="' . $alt . '" title="' . $alt . '" class="latex" width="' . $width . '"  height="' . $height . '" srcset="' . $url . '&amp;zoom=2 2x" scale="2" />';
+	}
+
+	/**
+	 * The shortcode way. The attributes are the same as the old ones - 'fg' and 'bg', instead of foreground
+	 * and background, and 's' is for the font size.
+	 *
+	 * @since 1.1
+	 *
+	 * Example: [latex s=4 bg=00f fg=ff0]\LaTeX[/latex]
+	 *
+	 * @param array   $atts
+	 * @param string  $content Default ''.
+	 * @return string LaTeX expression.
+	 */
+	public function latex_shortcode( $atts, $content = '' ) {
+		extract( shortcode_atts( array(
```

Comment:
:+1: Totally agree. I initially refrained from messing with this because the WordPress PHP Coding Standard says,

> It is permissible to space out the specific action or filter lines being documented to meet the coding standards, but do not refactor other code in the file.

Since I moved all the code into a class these lines ended up being modified anyway so I'll be very very happy to refactor that now. :)